### PR TITLE
[WIP] CB-193: Gulp for LESS and JS

### PR DESCRIPTION
**Pull request by opatel99**

### CODE REVIEW

File: critiquebrainz/frontend/templates/base.html [hunk 1]
```diff
@@ -9,18 +9,12 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <link rel="icon" href="{{ url_for('static', filename='img/favicon.png') }}" type="image/x-icon"/>
-
-    {% if config.COMPILE_LESS %}
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/base.html]: Don't forget to remove this variable from config file, since it's no longer used.

Comment (opatel99) [inline critiquebrainz/frontend/templates/base.html]: Removed COMPILE_LESS variable in bde7ff9627199dd7ddffeb8f2dd35b5c2822bff0

**Comment by gentlecat** [comment]
Can you fix indentation and formatting in new JavaScript files? Also, please make sure files end with a [newline](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

File: critiquebrainz/frontend/static/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    b.external(commonBundle);
+  });
+  var sharingBundle = runYarb('sharing.js');
+  var spotifyBundle = runYarb('spotify.js', function (b) {
+    b.external(commonBundle);
+  });
+
+
+
+
```
Comment (gentlecat) [inline critiquebrainz/frontend/static/gulpfile.js]: Why is there so much whitespace?

File: critiquebrainz/frontend/templates/event.html [hunk 3]
```diff
@@ -38,23 +38,7 @@ <h2 class="pull-left">
...
-
-        var popup = L.popup()
-          .setContent("<b>" + {{ _("Held At") | tojson }} + "</b><br> " + {{ place['name'] | tojson }});
-
-        var marker = L.marker([{{ lat }}, {{ long }}])
-          .addTo(map)
-          .bindPopup(popup)
-          .openPopup();
-      </script>
+      <script src="{{ 'edit.js'| make_static_path }}"></script>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/event.html]: Be consistent. Either have spaces around `|` or don't, but stick to the same style.

File: critiquebrainz/frontend/templates/sharing.html [hunk 4]
```diff
@@ -3,36 +3,19 @@
...
-        var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
-        po.src = 'https://apis.google.com/js/platform.js';
-        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
-      })();
-    </script>
   </li>
 
 </ul>
+
+<script src="{{ 'sharing.js'|make_static_path }}">
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/sharing.html]: These scripts shouldn't be moved either.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "jquery": "^2.1.4",
+    "leaflet": "^0.7.7",
+    "less": "^2.5.3",
+    "less-plugin-clean-css": "^1.5.1",
+    "lodash": "^3.10.1",
+    "path": "^0.12.7",
+    "q": "^1.4.1",
+    "through2": "^2.0.0",
+    "vinyl-source-stream": "^1.1.0",
+    "yarb": "^0.6.1"
```
Comment (gentlecat) [inline package.json]: It's better to reference specific versions of all packages.

**Comment by gentlecat** [comment]
It would be better to convert all calls to `url_for` for static files to use `make_static_path` filter.

**Comment by gentlecat** [comment]
Also CB-111 is completely unrelated to this.

File: critiquebrainz/frontend/static/js/templateScripts/sharing.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        if (d.getElementById(id)) return;
+        js = d.createElement(s); js.id = id;
+        js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.0";
+        fjs.parentNode.insertBefore(js, fjs);
+      }(document, 'script', 'facebook-jssdk'));
+      (function() {
+        var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
+        po.src = 'https://apis.google.com/js/platform.js';
+        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
+      })();
```
Comment (gentlecat) [inline critiquebrainz/frontend/static/js/templateScripts/sharing.js]: This should stay on a page.

File: critiquebrainz/frontend/static/js/templateScripts/searchSelector.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        })
+        .done(function(data) {
+          loading_message.hide();
+          $("#results").append(data.results);
+          if (data.more === true) more_button.show();
+        })
+        .fail(function() {
+          alert('{{ _("Failed to load more search results!") | tojson }}');
+        });
+      }
```
Comment (gentlecat) [inline critiquebrainz/frontend/static/js/templateScripts/searchSelector.js]: You forgot to fix indentation here and in other files.

**Comment by gentlecat** [comment]
I finished fixing this.

**Pull request closed without merging**